src/Chapter04.Tests/Listing04.50.Tests.cs

```diff
@@ -3,16 +3,55 @@
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter04.Listing04_50.Tests
 {
     [TestClass]
-    public class TicTacToeTests
+    public class ProgramTests
     {
         [TestMethod]
-        public void Main_BreakOnPlayerWinning_Player1Winner()
+        public void ValidateAndMove_Input4_ValidInput()
         {
             const string expected =
-                @"Player 1 was the winner";
+                @"";
+
+            int[] playerPositions = new int[1];
+            int currentPlayer = 0;
+            string input = "4";
+
+            IntelliTect.TestTools.Console.ConsoleAssert.Expect(
+                expected,
+                _ => Program.ValidateAndMove(playerPositions, currentPlayer, input),
+                true);
+        }
+        
+        [TestMethod]
+        public void ValidateAndMove_Input0_InvalidInput()
+        {
+            const string expected =
+@"
+ERROR:  Enter a value from 1-9. Push ENTER to quit";
+
+            int[] playerPositions = new int[1];
```

Comment:
```suggestion
            int[] playerPositions = new[1];
```

Reply:
I don't think this compiles.

Reply:
Yea sorry not sure what I was thinking here.